Feat/edit site

andriacap opened the pull request:
- [x] Résoudre le problème des informations non update alors que l'objet envoyé contient bien les nouvelles infos
- [x] Résoudre le problème des types de site non chargé lorsqu'on "annule" l'édition de site (obliger de recharger la page pour avoir les types de site qui apparraissent) 

Closes #54

andriacap committed:
feat: edit site (init value btn-select)

- Change button select to accept incoming intial values
- Change visit component to add btn-select and form-g inside html

- WIP: update form component when initial values is up to date

Reviewed-by: andriac
[Refs_ticket] :  #5 , #6

andriacap committed:
feat: get information when edit site

- Get all fields from specific site.json into editform
- Fix problem redirection if edit object site

- WIP : check how to update specific fields from object
- WIP : check how to manage listOption types site when reload or when
  come back into component after first init

Reviewed-by: andriac
[Refs_ticket]: #5 , #6

andriacap committed:
fix: forgot add apiService property binding

Reviewed-by:andriac

andriacap committed:
feat: update site with good properties

Change types_site object to array of ids
Remove extra key "dataComplement" to assign config

Reviewed-by:andriac
[Refs_ticket]: #5 , #6

andriacap committed:
feat: prevent form appear if no type-site

- Hide form if btn-list type-site not selected
- Add custom error message mat-error if not selected (directive + custom
  message)

WIP: error message is showing up only if not selected after touched .
Maybe need to use asyncValidator ?

Reviewed-by: andriac
[Refs_tickets]: #5 , #6 , #54

andriacap committed:
feat: change order to emit event bEdit

- Change order between change formService.changeData and bEditChange
- Preprocess for types_site seems to be useless (see todo)

Reviewed-by: andriac
[Refs_ticket]: : #5 , #6 , #54

mvergez commented on backend/gn_module_monitoring/monitoring/objects.py:
```diff
@@ -27,11 +27,17 @@ class MonitoringSite(MonitoringObjectGeom):
+                    type_site_ids.append(id_type_site)
+            data["id_nomenclature_type_site"]=data["types_site"][0]['id_nomenclature_type_site']
+            #TODO: A enlever une fois qu'on aura enelever le champ "id_nomenclature_type_site" du model et de la bdd
+        
```
Tester de supprimer toutes ces lignes sauf `id_nomenclature_type_site` dans les 2 cas if/else
En attendant de supprimer `id_nomenclature_type_site`

mvergez commented on backend/gn_module_monitoring/routes/site.py:
```diff
@@ -144,3 +144,16 @@ def post_sites():
+
+
+@blueprint.route("/sites/<int:_id>", methods=["PATCH"])
+def pacth_sites(_id):
```
typo pacth => patch

mvergez commented on frontend/app/components/monitoring-visits/monitoring-visits.component.ts:
```diff
@@ -114,4 +134,60 @@ export class MonitoringVisitsComponent extends MonitoringGeomComponent implement
+      }
+    }
+    for(var k in this.site.data) this.site[k]=this.site.data[k];
+    this.site["types_site"] = this.config.types_site
```
this.site.types_site ?

mvergez commented on frontend/app/components/monitoring-visits/monitoring-visits.component.ts:
```diff
@@ -114,4 +134,60 @@ export class MonitoringVisitsComponent extends MonitoringGeomComponent implement
+        }
+      }
+    }
+    for(var k in this.site.data) this.site[k]=this.site.data[k];
```
const plutôt que var

mvergez commented on frontend/app/components/monitoring-visits/monitoring-visits.component.ts:
```diff
@@ -114,4 +134,60 @@ export class MonitoringVisitsComponent extends MonitoringGeomComponent implement
+  }
+
+  updateForm(){
+    this.site['specific'] = {};
```
this.site.specific et donc changer l'interface ?

mvergez commented on frontend/app/interfaces/object.ts:
```diff
@@ -23,3 +24,5 @@ export type SelectObject = {
 };
 
+
+export type IExtraForm = {frmCtrl :AbstractControl,frmName:string}
```
prettier là dessus

mvergez commented:
Closes #54

andriacap committed:
feat: solve request changes

- Apply prettier
- Change object key [''] by type properties
- Remove unused preprocess_data lines

[Refs_ticket] : closes #54

andriacap reviewed (commented):
OK pour les changements
